matrix_client/api.py

```diff
@@ -97,20 +97,15 @@ def validate_certificate(self, valid):
         self.validate_cert = valid
         return
 
-    def register(self, login_type, **kwargs):
+    def register(self, content={}, query_params={}):
         """Performs /register.
 
         Args:
-            login_type(str): The value for the 'type' key.
-            **kwargs: Additional key/values to add to the JSON submitted.
+            content(dict): The request payload. Should include "type" such as "m.login.password" for all non-guest registrations.
```

Comment:
If I'm reading the https://matrix.org/docs/spec/client_server/r0.2.0.html#post-matrix-client-r0-register API correctly, it now uses auth.type instead of type. Synapse might still accept type, but it shouldn't according to the spec.